tests/src/test/scala/org/apache/openwhisk/core/controller/test/ActionsApiTests.scala

```diff
@@ -194,27 +194,49 @@ class ActionsApiTests extends ControllerTestCommon with WhiskActionsApi {
     implicit val tid = transid()
     val action = WhiskAction(namespace, aname(), jsDefault("??"), Parameters("x", "b"))
     put(entityStore, action)
+
     Get(s"$collectionPath/${action.name}") ~> Route.seal(routes(creds)) ~> check {
       status should be(OK)
       val response = responseAs[WhiskAction]
       response should be(action)
     }
   }
 
-  it should "get action by name in explicit namespace" in {
+  it should "get action with `updated` field" in {
     implicit val tid = transid()
+
     val action = WhiskAction(namespace, aname(), jsDefault("??"), Parameters("x", "b"))
     put(entityStore, action)
+
+    // `updated` field should be compared with a document in DB
+    val a = get(entityStore, action.docid, WhiskAction)
+
+    Get(s"/$namespace/${collection.path}/${action.name}?code=false") ~> Route.seal(routes(creds)) ~> check {
+      status should be(OK)
+      val responseJson = responseAs[JsObject]
+      responseJson.fields("updated").convertTo[Long] should be(a.updated.toEpochMilli)
+    }
+
     Get(s"/$namespace/${collection.path}/${action.name}") ~> Route.seal(routes(creds)) ~> check {
       status should be(OK)
-      val response = responseAs[WhiskAction]
-      response should be(action)
+      val responseJson = responseAs[JsObject]
+      responseJson.fields("updated").convertTo[Long] should be(a.updated.toEpochMilli)
     }
+  }
 
-    // it should "reject get action by name in explicit namespace not owned by subject" in
-    val auser = WhiskAuthHelpers.newIdentity()
-    Get(s"/$namespace/${collection.path}/${action.name}") ~> Route.seal(routes(auser)) ~> check {
-      status should be(Forbidden)
+  it should "not update action `updated` field with a put" in {
+    implicit val tid = transid()
+
+    val action = WhiskAction(namespace, aname(), jsDefault(""))
+    val dummyUpdated = 1569400000000L;
+    val content = JsObject(
+      "exec" -> JsObject("code" -> "".toJson, "kind" -> action.exec.kind.toJson),
+      "updated" -> dummyUpdated.toJson)
+
+    Put(s"$collectionPath/${action.name}", content) ~> Route.seal(routes(creds)) ~> check {
+      status should be(OK)
+      val response = responseAs[WhiskAction]
+      response.updated.toEpochMilli should not be (dummyUpdated)
```

Comment:
The time stamp should also be greater than a time stamp you can create just before doing the api request. This way you can at least lower bound the value and make sure it’s assigned reasonably.

Reply:
I've changed the test code to create a timestamp before doing the API request.
And it checks that timestamp of response is greater than a timestamp which is created before.

```scala
val dummyUpdated = WhiskEntity.currentMillis().toEpochMilli
```

```scala
response.updated.toEpochMilli should be > dummyUpdated
```